main.cpp

```diff
@@ -115,17 +126,20 @@ struct <#structname2#>
 
 int main()
 {
-    T <#name1#>( , );                                             //6
-    T <#name2#>( , );                                             //6
+    T A( 32, "Ernie" );                                             //6
+    T B( 35, "Bert" );                                             //6
     
-    <#structName1#> f;                                            //7
-    auto* smaller = f.compare( , );                              //8
-    std::cout << "the smaller one is << " << smaller->name << std::endl; //9
+    F f;                                            //7
+    auto* smaller = f.compare( &A, &B );                              //8
+    if(smaller != nullptr)
+    {
+        std::cout << "the smaller one is << " << smaller->name << std::endl; //9
+    }
```

Comment:
add an `else` block that explains why `smaller` might be a `nullptr`
There are 2 reasons.  the `std::cout` statement in the `else` block should explain both reasons.
- [ ] revise